napi-inl.h

```diff
@@ -2498,12 +2498,66 @@ inline Error::Error(napi_env env, napi_value value) : ObjectReference(env, nullp
   if (value != nullptr) {
     napi_status status = napi_create_reference(env, value, 1, &_ref);
 
+    // Creates a wrapper object containg the error value (primitive types) and
+    // create a reference to this wrapper
+    if (status != napi_ok) {
+      napi_value wrappedErrorObj;
+      status = napi_create_object(env, &wrappedErrorObj);
+
+      NAPI_FATAL_IF_FAILED(status, "Error::Error", "napi_create_object");
+
+      status = napi_set_property(env,
+                                 wrappedErrorObj,
+                                 String::From(env, "errorVal"),
+                                 Value::From(env, value));
+      NAPI_FATAL_IF_FAILED(status, "Error::Error", "napi_set_property");
+
+      status = napi_set_property(
+          env,
+          wrappedErrorObj,
+          String::From(env,
+                       "4b3d96fd-fb87-4951-a979-eb4f9d2f2ce9-isWrapObject"),
+          Value::From(env, value));
+      NAPI_FATAL_IF_FAILED(status, "Error::Error", "napi_set_property");
+
+      status = napi_create_reference(env, wrappedErrorObj, 1, &_ref);
+    }
+
     // Avoid infinite recursion in the failure case.
     // Don't try to construct & throw another Error instance.
     NAPI_FATAL_IF_FAILED(status, "Error::Error", "napi_create_reference");
   }
 }
 
+inline Object Error::Value() const {
+  if (_ref == nullptr) {
+    return Object(_env, nullptr);
+  }
+
+  napi_value refValue;
+  napi_status status = napi_get_reference_value(_env, _ref, &refValue);
+  NAPI_THROW_IF_FAILED(_env, status, Object());
+
+  // We are checking if the object is wrapped
+  bool isWrappedObject = false;
+  napi_has_property(
```

Comment:
Maybe add a typecheck first to see that it's not a Symbol. If the reference is a Symbol it's definitely not wrapped. In new Node.js versions, the reference can be a symbol.